app/controllers/course/survey/surveys_controller.rb

```diff
@@ -84,7 +84,8 @@ def survey_params
       :title, :description, :base_exp, :time_bonus_exp, :start_at, :bonus_end_at, :end_at,
       :published, :allow_response_after_end, :allow_modify_after_submit
     ]
-    fields << :anonymous if !@survey.anonymous || !@survey.has_student_response?
+    fields << :anonymous if action_name == 'create' || !@survey.anonymous ||
```

Comment:
Addressed in #2280, but I think it might be better not to make it a validation that is run each time a survey is updated. 

There is already validation on the frontend, so we don't expect the `anonymous` param to be passed in the update request - for correctness we should check the toggle-ability when whitelisting params. Don't think we need to check it again and don't think there is another place where we should check for this 'invalid' transition (the states themselves are all valid).